### PR TITLE
Samsonova/send cv file

### PR DESCRIPTION
**Pull request by AlisaSamsonova**

### CODE REVIEW

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
+  Feature: Send CV
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: You've done a huge work! And your tests are great! So I only have a few comments))

About the very beginning of this file: you need to specify the file number. For example, 1.2.1 Send CV. feature. 
(if you want to understand how to name a file and scripts, you can see an example in the main repository (it was created here by Olga) and in the main branch in your Visual studio).

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  Scenario: positive sending CV
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: For each scenario, you need to specify its number. For example, 1.2.1_01 - the first one, 1.2.1_02 - the second one etc.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  Given form "Send CV" with blank fields
+  When I enter valid data in the field "Name"
+  And I enter valid data in the field "Email"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Usually these steps are written from the user's point of view. Maybe it might be smth like that: user enters valid data in the field "Name".

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  And I attach CV
+  And I ticked "I agree to the personal data processing"
+  And I click button "Send"
+  Then my CV has been send
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Perhaps we also can write some more result that we can see? For example: error message or indicator of some colour.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+  Scenario: exceeding the limit of characters in the field "Name"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Also here you need to add the script number.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  Given form "Send CV" with blank fields
+  When I enter 71 characters with space delimiter in the field "Name"
+  Then a message appears "The maximum number of characters is 70"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: In this line (I mean, in the step Then) we usually use the word "should": a message should appear "....."

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  Scenario: entering the number format data in the field "Name"
+  Given form "Send CV" with blank fields
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: The same thing like before: it needs a number of scenario. 
And also in the step Given we use past tense, for example: user opened the form "Send CV"

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  Scenario: upload a file larger than 10 Mb
+  Given form "Send CV" with blank fields
+  When I attach 11 Mb CV-file 
+  Then an error message appears
+
+  
+  Scenario: don't tiked "I agree to the personal data processing"
+  Given form "Send CV" with blank fields
+  When I don't tiked "I agree to the personal data processing"
+  Then "I agree to the personal data processing" changes color to red
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: 1) We can use quotation marks ("") if we know that this text message might change its colour: for example - "red" colour or "green" colour.
2) in the step Then we usually use the word "should" in order to describe the expected result.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  Then a message appears "Invalid name format"
+
+
+  Scenario: entering invalid data in the field "Email"
+  Given form "Send CV" with blank fields
+  When I enter email-adress without "@"-character in the field "Email"
+  Then a message appears "This field is required"
+
+
+  Scenario: upload a file larger than 10 Mb
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Perhaps you can add more test cases, for example to check format of attached CV file (positive and negative options).

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 1 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Thank you)

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 2 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 3 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 4 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 5 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 6 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 7 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 9 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Oh, no, less is more :)

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature (hunk 8 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  Given form "Send CV" with blank fields
+  When I attach 11 Mb CV-file 
+  Then an error message appears
+
+  
+  Scenario: don't tiked "I agree to the personal data processing"
+  Given form "Send CV" with blank fields
+  When I don't tiked "I agree to the personal data processing"
+  Then "I agree to the personal data processing" changes color to red
+
```
Comment (AlisaSamsonova) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/send_cv.feature]: Thank you for comments)
Now i will try to push it (May the force be with me :) )

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 11]
```diff
@@ -0,0 +1,40 @@
+  Feature: 1.1.Send CV
+
+  Scenario: 1.1.01 positive sending CV
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: In the name of each scenario, we need to use an underscore: for example 1.1_01, next one: 1.1_02
As in the screenshot
image

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  Scenario: 1.1.04 entering invalid data in the field "Email"
+  Given user opened the form "Send CV"
+  When user enters email-address without "@"-character in the field "Email"
+  Then a message should appear "This field is required"
+
+
+  Scenario: 1.1.05 upload a file larger than 10 Mb
+  Given user opened the form "Send CV"
+  When user attaches 11 Mb CV-file
+  Then an error message should appear
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: I guess, that in the step Then we need to write the exact text of the error. According to Olga's message 


image

Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 11 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 13]
```diff
@@ -0,0 +1,40 @@
+  Feature: 1.1.Send CV
+
+  Scenario: 1.1_01 positive sending CV
+  Given user opened the form "Send CV"
+  When user enters valid data in the field "Name"
+  And user enters valid data in the field "Email"
+  And user attaches CV
+  And user ticked "I agree to the personal data processing"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: And user marks (I guess, because in the step When we don't use Past tense)

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+  And user click button "Send"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: And user clicks (or pushes) the "Send" button

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+  Scenario: 1.1_05 upload a file larger than 10 Mb
+  Given user opened the form "Send CV"
+  When user attaches 11 Mb CV-file
+  Then an error message should appear "Something went wrong Max file size 10 Mb"
+
+  
+  Scenario: 1.1_06 don't ticked "I agree to the personal data processing"
+  Given user opened the form "Send CV"
+  When user didn't ticked "I agree to the personal data processing"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: In the step When we need to say about user's actions and not in the Past. For example, user not marks

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+  Then "I agree to the personal data processing" changes color to "red"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: In the step Then we usually use the word "should" as a result what we expect here.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 13 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: ok

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 14 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: ok

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 16 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 15 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: ok, but i will write "does not mark"

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+  Then a message should appear "This field is required"
+
+
+  Scenario: 1.1_05 upload a file larger than 10 Mb
+  Given user opened the form "Send CV"
+  When user attaches 11 Mb CV-file
+  Then an error message should appear "Something went wrong Max file size 10 Mb"
+
+  
+  Scenario: 1.1_06 don't ticked "I agree to the personal data processing"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: Sorry, I missed it before - all this text ("I agree to ......") would be better call it as a checkbox, according to links which was sent us by Olga. Smth like that: Checkbox isn't mark

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+  Then a message should appear "Invalid name format"
+
+
+  Scenario: 1.1_04 entering invalid data in the field "Email"
+  Given user opened the form "Send CV"
+  When user enters email-address without "@"-character in the field "Email"
+  Then a message should appear "This field is required"
+
+
+  Scenario: 1.1_05 upload a file larger than 10 Mb
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:30]: Maybe it will be better in order to testing coverage to add one more test - for example, positive one: attach a CV file in valid size (10 Mb or less)

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+  Given user opened the form "Send CV"
+  When user attaches 11 Mb CV-file
+  Then an error message should appear "Something went wrong Max file size 10 Mb"
+
+  
+  Scenario: 1.1_06 don't ticked "I agree to the personal data processing"
+  Given user opened the form "Send CV"
+  When user does not mark "I agree to the personal data processing"
+  Then "I agree to the personal data processing" should change color to "red"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: Perhaps it will be  better to write like this: checkbox should change colour to "red"? Because all this thing is called "checkbox".

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 20]
```diff
@@ -0,0 +1,40 @@
+  Feature: 1.1.Send CV
+
+  Scenario: 1.1_01 positive sending CV
+  Given user opened the form "Send CV"
+  When user enters valid data in the field "Name"
+  And user enters valid data in the field "Email"
+  And user attaches CV
+  And user marks "I agree to the personal data processing"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: We can write it like this: "user marks checkbox with agreement to the personal data processing", I guess

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+  And user clicks the "Send" button
+  Then my CV has been send and success message should appear
+
+
+  Scenario: 1.1_02 exceeding the limit of characters in the field "Name"
+  Given user opened the form "Send CV"
+  When user enters 71 characters with space delimiter in the field "Name"
```
Comment (AlexandraPopovaz) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:15]: I guess you can add some tests, for example with happy path (positive), because according to this task we need to use different test design techniques and it will be great to use their in that case.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 20 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 21 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:15]: thanks, no, I want to make ready tests better

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 18 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:30]: done, i added this in 1.1_01

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 17 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: done

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature (hunk 19 above)
Comment (AlisaSamsonova) [reply qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature]: done

**Review by AlexandraPopovaz — APPROVED** [review]

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 22]
```diff
@@ -0,0 +1,40 @@
+  Feature: 1.1.Send CV
+
+  Scenario: 1.1_01 positive sending CV
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:3]: We need to use specific test data. If you would like to input some text - you should write this text. 
If you would like to attach some files - you should attach this file. 

Valid for all scenarios.

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+  And user clicks the "Send" button
+  Then my CV has been send and success message should appear
+
+
+  Scenario: 1.1_02 exceeding the limit of characters in the field "Name"
+  Given user opened the form "Send CV"
+  When user enters 71 characters with space delimiter in the field "Name"
+  Then a message should appear "The maximum number of characters is 70"
+
+  Scenario: 1.1_03 entering the number format data in the field "Name"
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:18]: If you would like to check the validation of the field "Name" (or any other field) - please use Examples and different parameters (incorrect symbols, many symbols etc)

File: qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+  Then a message should appear "This field is required"
+
+
+  Scenario: 1.1_05 upload a file larger than 10 Mb
+  Given user opened the form "Send CV"
+  When user attaches 11 Mb CV-file
+  Then an error message should appear "Something went wrong Max file size 10 Mb"
+
+  
+  Scenario: 1.1_06 Checkbox isn't mark
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/samsonova_sendCV/1.1.send_cv.feature:36]: Small test coverage. You should add more scenarios or Scenario Outline to check all possible variants for the best test coverage.

**Review by shepelevaolga — CHANGES_REQUESTED** [review]
Feedback for Author: 
- Please use specific test data
- Extend test coverage 
- Didn't catch why all commits were done by another account. 

Feedback for the Reviewer: 
- your comments were valid, but you should insist on the best test coverage. Otherwise, if the author does not follows your suggestion - you should stay your comments unresolved and not mark this PR as Approved.
